docs/tutorials/getting-started-with-flask.rst

```diff
@@ -0,0 +1,114 @@
+Getting started with Rockcraft for Flask
+----------------------------------------
+
+In this tutorial you will learn how to create a Flask application, build a rock
+with the Flask application and run it.
+
+Prerequisites
+-------------
+- snap enabled system (https://snapcraft.io)
+- LXD installed (https://linuxcontainers.org/lxd/getting-started-cli/)
+- Docker installed (https://snapcraft.io/docker)
+- a text editor
+
+Create Flask application
+------------------------
+
+Create a new directory and write the following into a text editor and save it as
+``requirements.txt``:
+
+.. literalinclude:: code/getting-started-with-flask/requirements.txt
+
+Install ``python3-venv`` on your host and create a virtual environment:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:create-venv]
+    :end-before: [docs:create-venv-end]
+    :dedent: 2
+
+In the same directory, write the following into a text editor and save it as
+``app.py``:
+
+.. literalinclude:: code/getting-started-with-flask/app.py
+    :language: python
+
+Run and test the flask application to verify everything is working:
+
+.. literalinclude:: code/getting-started-with-flask/task.yaml
+    :language: bash
+    :start-after: [docs:run-flask]
+    :end-before: [docs:run-flask-end]
+    :dedent: 2
+
+The ``curl`` command should print ``Hello, world!``.
+
+Install Rockcraft
```

Comment:
some of the prerequisites are on top, and they are not needed for the previous section. Then there's also this requisite. The approach should be: either all in under a single step, or each one explained only when needed

Reply:
IMHO the prerequisites should be at the top, in a single steps, explained for what they are needed.

The reason for this is because readers might get confused with instructions like `lxd init --auto` and end up running that everytime they want to do a `rockcraft pack`

Reply:
Moved to the top and ensured all pre-requesits are included with an explanation